src/serialization/interval_log/mod.rs

```diff
@@ -365,7 +366,11 @@ impl<'a, 'b, W: 'a + io::Write, S: 'b + Serializer> InternalLogWriter<'a, 'b, W,
     }
 
     fn write_comment(&mut self, s: &str) -> io::Result<()> {
-        write!(self.writer, "#{}\n", s)
+        for l in s.split('\n') {
+            write!(self.writer, "#{}\n", l)?;
```

Comment:
Do you not still have to escape control characters? `\r` for example?

Reply:
Not unless we start treating `\r` as a line separator. I added a test that has every control character in a comment. It gets a new line with a leading `#` for the `\n` but all others are left alone, and it parses fine.